### PR TITLE
fix: handle YouTube URLs in any format to correctly extract video ID

### LINKED ISSUE #12: Songs not loading up on dashboard
When using the web interface, the songs don't load. They are temporarily added to the queue where its thumbnail doesnt seem to load as well.

### How to replicate :

Step 1: Log in
Step 2 : Go to the dashboard ( you are automatically redirected)
Step 3 : Try to load up any song from yt.
image

Step 4 : Click the "Add to Queue" button

image

Step 5 : wait for a few seconds. It does not queue up the song.

image

I'm new to this. Please inform me if this issue is redundant or irrelevant.

### PR DESCRIPTION
**Pull request by Manoo07**
Fixes #12 , #32 , #33
This PR adds :
- Addresses an issue where the application was unable to extract the YouTube video ID from URLs that did not match a specific format. 
- The previous implementation had limitations, causing failures when encountering alternate or non-standard YouTube URL structures, such as URLs containing `parameters`, `shortened links`, or `embedded formats`.

image

### CODE REVIEW

**Comment by SujithThirumalaisamy** [comment]
Can you please fix the lint

**Comment by SujithThirumalaisamy** [comment]
Ideally this is a few line changes. Please open a new PR without the lints

**Comment by Manoo07** [comment]
Sure

File: next-app/app/api/streams/route.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export async function POST(req: NextRequest) {
       );
     }
 
-    const extractedId = data.url.split("?v=")[1];
+    const extractedId = data.url.split("?v=")[1].split("&")[0];
```
Comment (SujithThirumalaisamy) [inline next-app/app/api/streams/route.ts:58]: This will fail for shorterned links like `https://youtu.be/dQw4w9WgXcQ`

Comment (Manoo07) [reply next-app/app/api/streams/route.ts]: Hey @SujithThirumalaisamy ,Thanks for pointing that out.
This should now correctly handle both standard and shortened YouTube URLs. Please test it with different URL formats and let me know if there are any other issues.

**Comment by SujithThirumalaisamy** [comment]
Merging this for now. Can fix if there are something more. I think of nothing

**Pull request merged**